marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -315,6 +301,96 @@ function getPrimaryValue(entityInstance, entityDefinition) {
   }
 }
 
+//Helper function to add properties to each result instance under results array in searchReponse
+function addEntitySpecificProperties(result, entityInfo, selectedPropertyMetadata) {
+  let instance = null;
+  let entityTitle = entityInfo.entityName;
+  result.entityProperties = [];
+
+  try {
+    instance = getEntityInstance(result.uri);
+  } catch (error) {
+    console.log(`Unable to obtain entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
+  }
+
+  if (instance != null) {
+    let entityDef = entityInfo.entityModel.definitions[entityTitle];
+    const entityInstance = instance[entityTitle];
+    if (!entityInstance) {
+      console.log(`Unable to obtain entity instance from document with URI '${result.uri}' and entity name '${entityTitle}'; will not add entity properties to its search result`);
+    } else {
+      selectedPropertyMetadata.forEach(parentProperty => {
+        result.entityProperties.push(getPropertyValues(parentProperty, entityInstance));
+        result.primaryKey = getPrimaryValue(entityInstance, entityDef);
```

Comment:
Pinging @rahulvudutala - I don't think this should be in the forEach loop, it should be outside the loop. 

To avoid duplication with the same code in addGenericEntityProperties, you can then make an "addPrimaryKeyToResult(entityInstance, entityDef)" function to reuse between the two functions.

Reply:
@rjrudin Agree with Rob. This is when an entityModel is selected. So the primary key and primaryKeyValue should be set for each result but not for each for each selectedMetadataProperty for each result

Reply:
Done.